src/main/java/org/cryptomator/jsmb/util/MemorySegments.java

```diff
@@ -30,10 +30,11 @@ public static MemorySegment copy(MemorySegment original) throws IllegalArgumentE
 	 * @throws IllegalArgumentException If the resulting MemorySegment is too large to create (backed by a byte[])
 	 */
 	public static MemorySegment concat(MemorySegment left, MemorySegment right) {
-		if (left.byteSize() + right.byteSize() > Integer.MAX_VALUE) {
+		long size = left.byteSize() + right.byteSize();
+		if (size > Integer.MAX_VALUE || size < 0) {
```

Comment:
for clarity, let's use `Math.addExact` to check for overflows

```suggestion
		long size = Math.addExact(left.byteSize() + right.byteSize());
		if (size > Integer.MAX_VALUE) {
```